frontend/src/components/QueryBuilder/QueryBuilder.tsx

```diff
@@ -1607,7 +1607,8 @@ function QueryBuilder(props: QueryBuilderProps) {
 				removeSelectedSegment()
 			}
 		}
-	}, [removeSelectedSegment, setDefault, selectedSegment, readonly])
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [])
```

Comment:
Fixes bugs where the query build was unmounting necessarily, causing the segment to be reset

Reply:
looks good in the reflame preview!